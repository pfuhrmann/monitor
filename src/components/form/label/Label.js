import React from 'react';
import PropTypes from 'prop-types';
import './Label.scss';

const Label = ({ children, htmlFor }) => (
  <label htmlFor={htmlFor}>{children}</label>
);

Label.propTypes = {
  children: PropTypes.node,
  htmlFor: PropTypes.string,
};

Label.defaultProps = {
  children: '',
  htmlFor: null,
};

export default Label;
