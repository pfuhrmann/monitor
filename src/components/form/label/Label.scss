@import "../../../variables.scss";

label {
  color: $color-gulf-blue;
  font-weight: $font-weight-semi-bold;
}
