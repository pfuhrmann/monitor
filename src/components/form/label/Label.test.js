import React from 'react';
import ReactDOM from 'react-dom';
import Label from './Label';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Label htmlFor="field">Some label</Label>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders label correctly', () => {
  const component = renderer.create(
    <Label htmlFor="field">Some label</Label>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
