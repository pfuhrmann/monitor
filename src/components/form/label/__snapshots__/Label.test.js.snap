// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders label correctly 1`] = `
<label
  htmlFor="field"
>
  Some label
</label>
`;
